MaxText/inference_microbenchmark.py

```diff
@@ -274,8 +274,17 @@ def main(config):
       config, engine, params, decode_state, engine.max_concurrent_decodes, cache_size, model_size, benchmark_loop_iters)
 
   results = collate_results(config, benchmark_results, model_size, cache_size, num_model_params)
+  if config.inference_microbenchmark_flatten_results:
+    flattened_results = {}
+    for key, value in results.items():
+      if isinstance(value, dict):
+        flattened_results.update(value)
+      else:
+        flattened_results[key] = value
+    results = flattened_results
   write_results(results, filename=config.inference_microbenchmark_log_file_path)
```

Comment:
Feel free to modify this function for dimension_json!

Reply:
Didn't refactor to utils for now.